core/src/main/java/com/digitalpebble/storm/crawler/bolt/FetcherBolt.java

```diff
@@ -336,42 +324,61 @@ public synchronized FetchItemQueue getFetchItemQueue(String id) {
                 fiq = new FetchItemQueue(conf, customThreadVal, crawlDelay,
                         minCrawlDelay);
                 queues.put(id, fiq);
-
-                // Reset the cyclic iterator to start of the list.
-                it = Iterables.cycle(queues.keySet()).iterator();
             }
             return fiq;
         }
 
         public synchronized FetchItem getFetchItem() {
-
-            if (queues.isEmpty() || !it.hasNext())
+            if (queues.isEmpty()) {
                 return null;
+            }
 
             FetchItemQueue start = null;
 
             do {
-                FetchItemQueue fiq = queues.get(it.next());
+                Map.Entry<String, FetchItemQueue> nextEntry;
+                FetchItemQueue fiq;
+                Iterator i = queues.entrySet().iterator();
```

Comment:
I've just test both cases : 

Case 1 : 
```java
Iterator i = queues.entrySet().iterator();
Map<String, FetchQueue> tmpQueues = new LinkedHashMap<>();

try{
    do {
        // Stuff                
    
        // We remove the entry to reput at the end of the map
        i.remove();
        // Stuff
        tmpQueues.put(nextEntry.getKey(), nextEntry.getValue());
    
        // Stuff
    } while (i.hasNext());
    return null;
}finally {
    this.queues.putAll(tmpQueues);
}
```
Case 2 : 
```java
do {
    Iterator i = queues.entrySet().iterator();
    // Stuff                
    
    // We remove the entry to reput at the end of the map
    i.remove();
    // Stuff
    queues.put(nextEntry.getKey(), nextEntry.getValue());
    
    // Stuff
} while (!queues.isEmpty());
return null;
```

With 1000 Iterations, Case 2 is faster than one... about 10%